generators/requirefilegen/src/require_file_generator.rb

```diff
@@ -0,0 +1,232 @@
+# encoding: utf-8
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+
+require 'erb'
+require 'json'
+require_relative 'require_file_generator_options_parser'
+
+class RequireFileGenerator
+  attr_accessor :requires
+
+  def initialize(azure_sdk_location, mgmt_sdks_location)
+    @azure_sdk_location = azure_sdk_location
+    @mgmt_sdks_location = mgmt_sdks_location
+    @current_location = Dir.pwd
+    @file_to_be_written = ''
+    @requires = []
+  end
+
+  #
+  # Generates require files for the individual gems.
+  # These files must include the require for individual versions
+  # and individual profiles.
+  #
+  # Sample structure for an individual gem is as follows:
+  #     azure_mgmt_analysis_services
+  #        +----lib
+  #        |     +----2016-05-16
+  #        |     |        +------......
+  #        |     +----2017-07-14
+  #        |     |        +------......
+  #        |     +----profiles
+  #        |     |        +------common
+  #        |     |        +------profile_2016_05_16
+  #        |     |        +------profile_2017_07_14
+  #        |     |        +------profile_latest
+  #        |     +----azure_mgmt_analysis_services.rb (*)
+  #        |     +----module_definition.rb
+  #        |     +----version.rb
+  #        +----spec
+  #        |.....
+  # * - The file to be created.
+  #
+  # In the lib folder, the module definition and the version files need
+  # to be ignored.
+  #
+  # Within each version, there is a generated folder which has a single file
+  # that needs to be added to the require file.
+  #
+  # Within the profiles folder, the common folder needs to be ignored. Under the
+  # rest of the (profiles) folder, the module definition file needs to be ignored
+  # and the remaining one file needs to be added to the require file.
+  #
+  def generate_require_files_for_individual_gems
+    puts 'Generating require files for individual gems'
+    Dir.chdir("#{@current_location}/#{@mgmt_sdks_location}")
+    gems = Dir['*'].reject{|o| not File.directory?(o)}
+    gems.each do |gem|
+      # azure_mgmt_insights is a special case gem which we have stopped
+      # supporting. But the folder is present and should be ignored in this
+      # process
+      if(gem == 'azure_mgmt_insights')
```

Comment:
In general, I'd create an array of blacklist gems that we ignore. that way we just keep adding to that list instead of checking for each one.

Reply:
Done